jujugui/static/gui/src/app/components/shared/status/status/status.js

```diff
@@ -0,0 +1,244 @@
+        model={model}
+        statusFilter={this.state.statusFilter} />);
+  }
+
+  /**
+    Generate the remote applications fragment of the status.
+    @returns {Object} The resulting element.
+  */
+  _generateRemoteApplications() {
+    const {remoteApplications} = this.props.entities;
+    if (remoteApplications.length === 0) {
+      return null;
+    }
+    return (
+      <StatusRemoteApplicationList
+        remoteApplications={remoteApplications}
+        statusFilter={this.state.statusFilter} />);
+  }
+
+  /**
+    Generate the applications fragment of the status.
+    @returns {Object} The resulting element.
+  */
+  _generateApplications() {
+    const {applications} = this.props.entities;
+    if (applications.length === 0) {
+      return null;
+    }
+    return (
+      <StatusApplicationList
+        applications={applications}
+        generateApplicationOnClick={this.props.generateApplicationOnClick}
+        generateApplicationURL={this.props.generateApplicationURL}
+        generateCharmURL={this.props.generateCharmURL}
+        onCharmClick={this.props.navigateToCharm}
+        statusFilter={this.state.statusFilter} />);
+  }
+
+  /**
+    Generate the units fragment of the status.
+    @returns {Object} The resulting element.
+  */
+  _generateUnits() {
+    const {units} = this.props.entities;
+    if (units.length === 0) {
+      return null;
+    }
+    return (
+      <StatusUnitList
+        applications={this.props.entities.applications}
+        generateMachineURL={this.props.generateMachineURL}
+        generateUnitOnClick={this.props.generateUnitOnClick}
+        generateUnitURL={this.props.generateUnitURL}
+        onMachineClick={this.props.navigateToMachine}
+        statusFilter={this.state.statusFilter}
+        units={units} />);
+  }
+
+  /**
+    Generate the machines fragment of the status.
+    @returns {Object} The resulting element.
+  */
+  _generateMachines() {
+    const {machines} = this.props.entities;
+    if (machines.length === 0) {
+      return null;
+    }
+    return (
+      <StatusMachineList
+        generateMachineOnClick={this.props.generateMachineOnClick}
+        generateMachineURL={this.props.generateMachineURL}
+        machines={machines}
+        statusFilter={this.state.statusFilter} />);
+  }
+
+  /**
+    Generate the relations fragment of the status.
+    @returns {Object} The resulting element.
+  */
+  _generateRelations() {
+    const {relations} = this.props.entities;
+    if (relations.length === 0) {
+      return null;
+    }
+    return (
+      <StatusRelationList
+        applications={this.props.entities.applications}
+        generateApplicationURL={this.props.generateApplicationURL}
+        onApplicationClick={this.props.navigateToApplication}
+        relations={relations}
+        statusFilter={this.state.statusFilter} />);
+  }
+
+  render() {
+    let content;
+    if (!this.props.entities.model.modelUUID) {
```

Comment:
This should check that model is defined as it may not be if we aren't connected.